src/NexusMods.Paths/AbsolutePath.cs

```diff
@@ -84,19 +85,34 @@ internal AbsolutePath(string? directory, string fileName, IFileSystem fileSystem
     /// </remarks>
     public static AbsolutePath FromFullPath(string fullPath, IFileSystem? fileSystem = null)
     {
-        var index = fullPath.LastIndexOf(Path.DirectorySeparatorChar);
-        if (index < 0)
-        {
-            // No directory
+        var span = fullPath.AsSpan();
+
+        // path is not rooted
+        var rootLength = GetRootLength(span);
+        if (rootLength == 0)
             return new AbsolutePath(null, fullPath, fileSystem ?? Paths.FileSystem.Shared);
```

Comment:
I have a question about this: Would a non-rooted path not be considered a non-full path; and thus be invalid use of the API?

Should we either throw [throwhelper] or use `Path.GetFullPath()` here?

Reply:
I think that's part of a bigger design issue with `AbsolutePath`: `Directory` can be null and `FileName` can be empty. There is also the interface `IPath` which also has a `FileName` property, which is implemented as `Path.GetFileName(FileName)`.

We should make a separate issue for this.

Reply:
Moves to issue https://github.com/Nexus-Mods/NexusMods.App/issues/189